src/cript/api/exceptions.py

```diff
@@ -129,6 +130,35 @@ def __str__(self) -> str:
         return error_message
 
 
+class CRIPTDuplicateNameError(CRIPTAPISaveError):
+    def __init__(self, api_response, json_data: str, parent_cript_save_error: CRIPTAPISaveError):
+        super().__init__(
+            parent_cript_save_error.api_host_domain, api_response["code"], api_response=api_response["error"], patch_request=parent_cript_save_error.patch_request, pre_saved_nodes=parent_cript_save_error.pre_saved_nodes, json_data=json_data
+        )
+
+        # We don't care if the data is invalid JSON
+        # So let's catch a couple of common exceptions and ensure still meaning error messages
+        # (and debug info in case it does happen.)
+        try:
+            json_dict = json.loads(self.json_data)
+        except (TypeError, json.JSONDecodeError):
+            self.name = "unknown_name"
+            self.node = "UnknownType"
+        try:
+            self.name = json_dict["name"]
+        except KeyError:
+            self.name = "unknown_name_key"
+        try:
```

Comment:
I wonder if there is a cleaner way of writing this or not

Reply:
Yes, there is.

You could use blanket style exception catching like this:

```python
try:
  json_dict = json.loads(self.json_data)
  self.name = json_dict["name"]
  self.node = json_dict["node"][0]
except (TypeError, json.JSONDecodeError, KeyError, IndexError):
  self.name = "unknown_name"
  self.node = "UnknownType"
```

But that is less granular and might not be the best possible error message.

Let me know what you prefer.

Reply:
hmm that is true. We can leave it how it was then for now and if we find a better way in the future we can refactor and upgrade it then. It has good comments so no one should be lost as to what is happening I think